include/vctr/Expressions/Core/Sqrt.h

```diff
@@ -0,0 +1,85 @@
+/*
+  ==============================================================================
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+    Copyright 2022- by sonible GmbH.
+
+    This file is part of VCTR - Versatile Container Templates Reconceptualized.
+
+    VCTR is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License version 3
+    only, as published by the Free Software Foundation.
+
+    VCTR is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+namespace vctr::Expressions
+{
+
+template <size_t extent, class SrcType>
+requires is::floatNumber<ValueType<SrcType>> || is::complexFloatNumber<ValueType<SrcType>>
+class Sqrt : ExpressionTemplateBase
+{
+public:
+    using value_type = ValueType<SrcType>;
+
+    VCTR_COMMON_UNARY_EXPRESSION_MEMBERS (Sqrt, src)
+
+    VCTR_FORCEDINLINE constexpr value_type operator[] (size_t i) const
+    {
+        VCTR_ASSERT (inputValueValid (src[i]));
+        return std::sqrt (src[i]);
+    }
+
+    //==============================================================================
+    // Platform Vector Operation Implementation
+    VCTR_FORCEDINLINE const value_type* evalNextVectorOpInExpressionChain (value_type* dst) const
+    requires is::suitableForAccelerateRealFloatVectorOp<SrcType, value_type, detail::dontPreferIfIppAndAccelerateAreAvailable>
+    {
+        const auto* s = src.evalNextVectorOpInExpressionChain (dst);
+        VCTR_ASSERT (std::all_of (s, s + size(), inputValueValid));
+
+        Expression::Accelerate::sqrt (s, dst, sizeToInt (size()));
+        return dst;
+    }
+
+    VCTR_FORCEDINLINE const value_type* evalNextVectorOpInExpressionChain (value_type* dst) const
+    requires is::suitableForIppRealOrComplexFloatVectorOp<SrcType, value_type, detail::preferIfIppAndAccelerateAreAvailable>
+    {
+        const auto* s = src.evalNextVectorOpInExpressionChain (dst);
+        VCTR_ASSERT (std::all_of (s, s + size(), inputValueValid));
+
+        Expression::IPP::sqrt (s, dst, sizeToInt (size()));
+        return dst;
+    }
+
+private:
+    /** If the value type is real, this checks if the value is greater or equal zero */
+    static constexpr bool inputValueValid (value_type in)
+    {
+        if constexpr (is::realNumber<value_type>)
+            return in >= value_type (0);
+
+        return true;
```

Comment:
The input value is always valid if `value_type` is not a real number?

Reply:
Yes, I think for complex numbers (which is basically all non-real numbers) the input is not restricted to any value range. Am I wrong with that assumption?

Reply:
I didn't expect complex numbers to be supported at all. If yes, maybe lets add a test case